### DISCUSSION TITLE
Saving Credentials in Remote Desktop

### DISCUSSION THREAD

**Discussion by brunodalopes** [BornToBeRoot/NETworkManager#1671] [Q&A] [answered]
Is anyway or setting to save credentials in rdp connections?
![image](https://user-images.githubusercontent.com/92134707/199158668-a5e61b32-07bc-4070-b815-c6ade60a959b.png)

**Comment by BornToBeRoot** [comment, marked as answer]
Hi @brunodalopes,

with the `Connect as` you can connect as another user with username + password.

If you want to save credentials you can do it in the "Group" dialog (if you want to use the credentials for multiple systems):

![image](https://user-images.githubusercontent.com/16019165/199306156-85b34b23-76db-46ff-8416-a995bad9e882.png)

Or in the "Profile" dialog:

![image](https://user-images.githubusercontent.com/16019165/199306308-7452f00e-97eb-4f11-ac44-4db5c7a2ff50.png)

This credentials will be used if you double click a profile or connect via right click `Connect`.

For multiple users on the same system you have to create multiple profiles (with a different name `System A - User A`, `System A - User B`)

---

If you don't set any credentials, the default credentials for `mstsc.exe` (`default.rdp` profile) is used.

---

**NOTE**

If you save the credentials in the group or profile, you should encrypt the profile file. Otherwise, they are stored in the profile file as plain text.


See: https://borntoberoot.net/NETworkManager/FAQ#how-to-enable-profile-file-encryption

**Comment by brunodalopes** [comment]
Thank you very much for the great software and the answer.

**Comment by brunodalopes** [comment]
Can you save passwords for putty as well?

**Reply by BornToBeRoot** [reply]
Theoretically yes, but these would then be displayed in the command line in the taskmanager. So for security reasons rather no. I would use KeePass (or something similar) for this.

If KeePass Auto-Type works / is set up in a normal PuTTY window, it also works in NETworkManager

**Comment by brunodalopes** [comment]
Nice. Thank you

**Reply by BornToBeRoot** [reply]
![KeePassAutoType](https://user-images.githubusercontent.com/16019165/199344685-8c15514e-fbcb-45db-9bc6-e87733000058.png)
Something like this works
